### PR TITLE
General consistent subtemplate names

### LINKED ISSUE #295: General - Deployment name generation in Subscription level deployments
Develop a consistent method of generating names for "Microsoft.Resources/Deployments" sections in arm templates.

### PR DESCRIPTION
**Pull request by deepend-dev**
closes #295

### CODE REVIEW

File: Cloudmarque.Azure/Public/Deploy/PaaS/function/New-CmAzPaasFunction.ps1 [hunk 1]
```diff
@@ -178,6 +181,8 @@ function New-CmAzPaasFunction {
 
                         $function.resourceGroupName = $functionAppSolution.generatedResourceGroupName
                         $function.planName = Get-CmAzResourceName -Resource "AppServicePlan" -Architecture "PaaS" -Location $function.location -Name $function.name
+
+                        $function.templateName = Get-CmAzResourceName -Resource "deployment" -Architecture "PaaS" -Location $function.location -Name "$commandName-con-$($function.name)"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/PaaS/function/New-CmAzPaasFunction.ps1:185]: is it always consumption?

File: Cloudmarque.Azure/Public/Deploy/PaaS/sql/New-CmAzPaasSql.ps1 [hunk 2]
```diff
@@ -165,6 +168,7 @@
 					"resourceDetails"    = @{
 						"family"                = $dbFamily
 						"sharedServer"          = $sharedServer
+						"templateName"			= $templateName
 						"type"                  = ($_.type ?? "").Tolower()
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/PaaS/sql/New-CmAzPaasSql.ps1:172]: missing ;

**Review by tm-luke — CHANGES_REQUESTED** [review]
Three things in addition to the comments in the review:

* Are we safely able to deploy without the deployment template names being too long?
* Have you ran the integration pipeline?
* Have you tested all cmdllets that are not in integration?

File: Cloudmarque.Azure/Public/Deploy/PaaS/function/New-CmAzPaasFunction.ps1 (hunk 1 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Deploy/PaaS/function/New-CmAzPaasFunction.ps1:185]: yes

File: Cloudmarque.Azure/Public/Deploy/PaaS/sql/New-CmAzPaasSql.ps1 (hunk 2 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Deploy/PaaS/sql/New-CmAzPaasSql.ps1:172]: this needs to be separate issue.. none of the properties in this command have ;

**Comment by deepend-dev** [comment]
> Three things in addition to the comments in the review:
> 
> 
> 
> * Are we safely able to deploy without the deployment template names being too long?
> 
> * Have you ran the integration pipeline?
> 
> * Have you tested all cmdllets that are not in integration?

- yes, limit of 64 applied.
- IP is almost done.
- I couldn't test WVD and shared gallery but that needs refactoring anyways

**Pull request merged**